cmd/acraserver/acraserver.go

```diff
@@ -175,7 +187,79 @@ func main() {
 		}()
 	}
 	if *withZone && !*disableHTTPApi {
-		go server.StartCommands()
+		if os.Getenv("_GRACEFUL_RESTART") == "true" {
+			go server.StartCommandsFromFileDescriptor(4)
+		} else {
+			go server.StartCommands(true)
+		}
+
+	}
+
+	log.Infof("PID: %v", os.Getpid())
+	if os.Getenv("_GRACEFUL_RESTART") == "true" {
+		go server.StartFromFileDescriptor(3)
+	} else {
+		go server.Start(true)
 	}
-	server.Start()
+
+	signal.Notify(SignalsChannel, syscall.SIGHUP, syscall.SIGTERM)
+	var WaitTimeoutError = errors.New("timeout")
+	for sig := range SignalsChannel {
+		if sig == syscall.SIGTERM {
+			// Stop accepting new connections
```

Comment:
I believe that code would be much readable if we divide this laaarge `if` handlers into separate functions.

like

```
func handleSigTerm(...)
func handleSigHup(...)
```

smaller, more readable

Reply:
Used our handler wrapper